generated/nidaqmx/_grpc_interpreter.py

```diff
@@ -3392,10 +3438,26 @@ def _assign_numpy_array(numpy_array, grpc_array):
     grpc_array_size = len(grpc_array)
     if isinstance(grpc_array, bytes):
         assert numpy_array.nbytes >= grpc_array_size
-        numpy_array.flat[:grpc_array_size] = numpy.frombuffer(grpc_array, dtype=numpy_array.dtype)
+        if numpy_array.ndim > 1:
+            (number_of_channels, number_of_samples_per_channel) = numpy_array.shape
+            read_array = numpy.frombuffer(grpc_array, dtype=numpy_array.dtype).reshape((number_of_channels, samples_read))
+            for row in range(0, number_of_channels):
+                for column in range(0, number_of_samples_per_channel):
+                    if column < samples_read:
+                        numpy_array[row, column] = read_array[row, column]
+        else:
+            numpy_array.flat[0:samples_read] = numpy.frombuffer(grpc_array, dtype=numpy_array.dtype)
     else:
         assert numpy_array.size >= grpc_array_size
-        numpy_array.flat[:grpc_array_size] = grpc_array
+        if numpy_array.ndim > 1:
+            (number_of_channels, number_of_samples_per_channel) = numpy_array.shape
+            read_array = numpy.array(grpc_array).reshape((number_of_channels, samples_read))
+            for row in range(0, number_of_channels):
+                for column in range(0, number_of_samples_per_channel):
+                    if column < samples_read:
+                        numpy_array[row, column] = read_array[row, column]
```

Comment:
Performance problems:
1. Looping over numpy arrays in Python is slow. Let numpy do the looping in C.
2. This creates a temporary numpy array, which I think we can avoid.

For 1, use slicing to create a view of a subset of the array (https://numpy.org/doc/stable/user/basics.indexing.html)
```
numpy_array[:, :samps_per_chan_read] = read_array
```

For 2, combine slicing with a flat iterator:
```
numpy_array[:, :samps_per_chan_read].flat[:grpc_array_size] = grpc_array
```

Note that `flat[:grpc_array_size]` handles the case where grpc_array is too short, preventing the flat iterator assignment from repeating the grpc_array elements. If samps_per_chan_read is consistent with grpc_array, then maybe this is not needed, but I'm not sure.